obspy/io/nordic/tests/test_nordic.py

```diff
@@ -0,0 +1,698 @@
+        self.assertEqual(_float_conv('1'), 1.0)
+        self.assertEqual(_str_conv(1), '1')
+        self.assertEqual(_int_conv('1.0256'), None)
+        self.assertEqual(_float_conv('1.0256'), 1.0256)
+        self.assertEqual(_str_conv(1.0256), '1.0256')
+
+    def test_read_wavename(self):
+        testing_path = os.path.join(self.testing_path, '01-0411-15L.S201309')
+        wavefiles = readwavename(testing_path)
+        self.assertEqual(len(wavefiles), 1)
+
+    def test_station_to_seisan(self):
+        t1 = UTCDateTime(2012, 3, 26)
+        t2 = UTCDateTime(2012, 4, 26)
+        client = Client('GEONET')
+        bulk = [('NZ', 'FOZ', '*', '*', t1, t2),
+                ('NZ', 'JCZ', '*', '*', t1, t2),
+                ('NZ', 'WVZ', '*', '*', t1, t2)]
+        inventory = client.get_stations_bulk(bulk, level="channel")
+        for station in inventory[0]:
+            sta_str = station_to_seisan(station)
+            # Simple check that the length is correct.
+            self.assertEqual(len(sta_str), 27)
+
+        for station in inventory[0]:
+            stat = station_to_seisan(station)
+            # Check that the lat and long are correct.
+            station_name = stat[1:6].strip()
+            lat = int(stat[6:8]) + (float(stat[8:13]) / 60)
+            if stat[13] == 'S':
+                lat *= -1
+            lon = int(stat[14:17]) + (float(stat[17:22]) / 60)
+            if stat[22] == 'W':
+                lon *= -1
+            elev = int(stat[23:])
+            self.assertEqual(station.code, station_name)
+            self.assertEqual(round(station.latitude, 2),
+                             round(lat, 2))
+            self.assertEqual(round(station.longitude, 2),
+                             round(lon, 2))
+            self.assertEqual(int(round(station.elevation)), elev)
+
+        inventory = client.get_stations_bulk(bulk)
+        # Check that it complains about trying to write the wrong type
+        for station in inventory[0]:
+            with self.assertRaises(IOError):
+                sta_str = station_to_seisan(station)
+
+    def test_read_event(self):
+        """Test the wrapper."""
+        testing_path = os.path.join(self.testing_path, '01-0411-15L.S201309')
+        event = read_nordic(testing_path)[0]
+        self.assertEqual(len(event.origins), 1)
+
+    def test_read_many_events(self):
+        testing_path = os.path.join(self.testing_path, 'select.out')
+        catalog = read_nordic(testing_path)
+        self.assertEqual(len(catalog), 50)
+
+    def test_write_select(self):
+        cat = read_events()
+        try:
+            write_select(cat, filename='select.out')
+            cat_back = read_events('select.out')
+            for event_1, event_2 in zip(cat, cat_back):
+                self.assertTrue(test_similarity(event_1=event_1,
+                                                event_2=event_2))
+        finally:
+            os.remove('select.out')
```

Comment:
Sorry, if I was cross-firing with another comment. But I think this is the most obspy-canonical handling of temp files in tests. It's always better to do the nitty gritty (i.e. handle the temp file removal) in one place (i.e. inside the implementation of `NamedTemporaryFile`) rather than duplicated across code (which is hard to maintain) and in less detail (`NamedTemporaryFile` checks if the file really exists before trying to remove it.. if something goes wrong in your code and the file isn't even created in the first place, your test will spit out some `IOError` (that needs debugging subsequently..)